### DISCUSSION TITLE
PHP 7.4 And Tidy-HTML 5.8

### DISCUSSION THREAD

**Discussion by LiamW** [shivammathur/homebrew-php#400] [Q&A / Help] [answered]
My tidy-html package got updated to 5.8 and now I can't load php 7.4 as it gives the error looking for 5.0

httpd: Syntax error on line 185 of /usr/local/etc/httpd/httpd.conf: Cannot load /usr/local/opt/php@7.4/lib/httpd/modules/libphp7.so into server: dlopen(/usr/local/opt/php@7.4/lib/httpd/modules/libphp7.so, 10): Library not loaded: /usr/local/opt/tidy-html5/lib/libtidy.5.dylib\n  Referenced from: /usr/local/opt/php@7.4/lib/httpd/modules/libphp7.so\n  Reason: image not found

dyld: Library not loaded: /usr/local/opt/tidy-html5/lib/libtidy.5.dylib
Referenced from: /usr/local/bin/php
Reason: image not found

I've tried reinstalling php 7.4 but no luck.

**Comment by shivammathur** [comment, marked as answer]
@LiamW 

Update: All formulae have been re-built. This should work now.

```zsh
brew update
brew reinstall shivammathur/php/php@7.4
```

**Reply by LiamW** [reply]
Thanks that was quick! All good now.

**Reply by zyrup** [reply]
Why do we have these issues

**Comment by arsyadarmawan** [comment]
I have run `brew reinstall php@7.2` but stil get error message `dyld: Library not loaded: /usr/local/opt/tidy-html5/lib/libtidy.5.dylib`

**Reply by shivammathur** [reply]
@arsyadarmawan

Please run the following commands 
1) `brew update`
2) `brew install tidy-html5`
3) `brew reinstall shivammathur/php/php@7.2 && brew link --overwrite --force php@7.2`

Make sure there are no errors on each step. If you get any errors, please open a new thread with complete logs.

**Reply by jamie-selesti** [reply]
I did `ln -s /usr/local/Cellar/tidy-html5/5.8.0/lib/libtidy.58.dylib /usr/local/Cellar/tidy-html5/5.8.0/lib/libtidy.5.dylib` to get the reference set up and then did `brew reinstall php@7.2` and everything worked

**Reply by shivammathur** [reply]
@jamie-selesti
If you reinstall from this tap `brew reinstall shivammathur/php/php@7.2`, this would not be required, after I had rebuilt all formulae https://github.com/shivammathur/homebrew-php/discussions/400#discussioncomment-996349, the PHP binaries use `/usr/local/opt/tidy-html5/lib/libtidy.58.dylib`. So reinstalling the PHP version and tidy should fix it.

**Reply by fransyrcc** [reply]
That works for me. Thank you @shivammathur

**Reply by andyscraven** [reply]
@jamie-selesti Saved a good half a day of my time, thanks.  The symlink (once I edited the path) fixed it for me.

**Comment by kossa** [comment]
For my case I fixed it by :
```
export PATH="/usr/local/opt/php@8.0/bin:$PATH"
export PATH="/usr/local/opt/php@8.0/sbin:$PATH"
```

**Reply by tbiswal** [reply]
This worked for me :) Thanks

**Reply by tonymra** [reply]
It really worked for me as well .

**Comment by rbenli** [comment]
On mac I just did 
`brew reinstall tidy-html5`
and it works
